internal/builders/maven/README.md

```diff
@@ -86,25 +88,51 @@ jobs:
 
 Now, when you invoke this workflow, the Maven builder will build both your artifacts and the provenance files for them.
 
-You can also release artifacts to Maven Central by adding the following step to your workflow:
+### Releasing directly to Maven Central
+
+You can also release artifacts to Maven Central with [the slsa-github-generator Maven publisher](https://github.com/slsa-framework/slsa-github-generator/blob/main/actions/maven/publish/action.yml) by adding the following step to your workflow:
 
 ```yaml
   publish:
-    needs: build
-    uses: slsa-framework/slsa-github-generator/.github/workflows/publish_maven.yml@v1.7.0
-    with:
-      provenance-download-name: "${{ needs.build.outputs.provenance-download-name }}"
-      provenance-download-sha256: "${{ needs.build.outputs.provenance-download-sha256 }}"
-      target-download-sha256: "${{ needs.build.outputs.target-download-sha256 }}"
-    secrets:
-      maven-username: ${{ secrets.OSSRH_USERNAME }}
-      maven-password: ${{ secrets.OSSRH_PASSWORD }}
-      gpg-key-pass: ${{ secrets.GPG_PASSPHRASE }}
-      gpg-private-key: ${{ secrets.GPG_PRIVATE_KEY }}
+    - name: publish
+      id: publish
+      uses: slsa-framework/slsa-github-generator/actions/maven/publish@v1.7.0
+      with:
+        provenance-download-name: "${{ needs.build.outputs.provenance-download-name }}"
+        provenance-download-sha256: "${{ needs.build.outputs.provenance-download-sha256 }}"
+        target-download-sha256: "${{ needs.build.outputs.target-download-sha256 }}"
+        maven-username: ${{ secrets.OSSRH_USERNAME }}
+        maven-password: ${{ secrets.OSSRH_PASSWORD }}
+        gpg-key-pass: ${{ secrets.GPG_PASSPHRASE }}
+        gpg-private-key: ${{ secrets.GPG_PRIVATE_KEY }}
 ```
 
 Now your workflow will build your artifacts and publish them to a staging repository in Maven Central.
 
+In the above example of the publish Action, the job that invokes the Maven builder is called `build`. The publish Action uses output from that job.
+
+#### Publisher requirements
+
+Besides adding the above workflow to your CI pipeline, you also need to add the following plugin to your `pom.xml`:
+
+```java
+<plugin>
+    <groupId>io.github.adamkorcz</groupId>
+    <artifactId>slsa-hashing-plugin</artifactId>
+    <version>0.0.1</version>
+    <executions>
+        <execution>
+            <goals>
+                <goal>hash-jarfile</goal>
```

Comment:
how do we know that `hash-jarfile` won't collide with existing files at runtime?

Reply:
> how do we know that `hash-jarfile` won't collide with existing files at runtime?

It refers to a goal in the Maven build cycle. It could collide with other plugin goal. I think Maven would complain in that case, but we can also make it more unique. WDYT?

Reply:
I see what you mean. It's already part of the `io.github.adamkorcz` namespace so I think you're right it should not collide.
@loosebazooka do you known more?

Reply:
Resolving since this should not be an issue. Feel free to reopen if you disagree.